src/consensus/blockstore/slot_block_data.rs

```diff
@@ -151,8 +152,8 @@ impl BlockData {
         // store last slice index, delete everything after last slice
         if is_last_slice && self.last_slice.is_none() {
             self.last_slice = Some(slice_index);
-            self.slices.split_off(&(slice_index + 1));
-            self.shreds.split_off(&(slice_index + 1));
+            self.slices.retain(|&ind, _| ind <= slice_index);
```

Comment:
we could also keep comparing the last index and popping.  Not sure what is more efficient.

Reply:
mh, too bad we can't use `split_off()` anymore, it would be O(log n), as opposed to O(n)...